### ISSUE TITLE
has_subwoofer throws KeyError: 'model_name'

### ISSUE THREAD

**Issue by eugen-vg** [SoCo/SoCo#960]
I'm looking to work with a subwoofer in the Arc+Sub setup and started from the simple example:
```python
from soco import SoCo
my_zone = SoCo('***')
print(my_zone.has_subwoofer)
```
It returns the error:

> Traceback (most recent call last):
>   File "main.py", line 3, in <module>
>     print(my_zone.has_subwoofer)
>   File "/.venv/lib/python3.9/site-packages/soco/core.py", line 518, in has_subwoofer
>     model_name = self.speaker_info["model_name"].lower()
> KeyError: 'model_name'

The code works fine if I run it under the debugger line by line. When I set a breakpoint to core.py at `model_name = self.speaker_info["model_name"].lower()`, I found that `self.speaker_info` was an empty collection until I didn't manually expand the properties of the `self` object in the debugger's watch. They were recalculated at that moment, which allowed the line to pass successfully. The `sub_gain` relies on `has_subwoofer` and also fails.

**Comment by eugen-vg** [comment]
I've made the `has_subwoofer` to work explicitly calling `my_zone.get_speaker_info()` before it, but I believe there should be a guard code inside `has_subwoofer` as for `is_soundbar`
```
if not self.speaker_info:
    self.get_speaker_info()
```

**Comment by pwt** [comment]
I agree. Thanks for spotting the issue and identifying the fix.

**Comment by pwt** [comment]
Fix released in SoCo v0.30.4.
